inbox/xep-sla.md

```diff
@@ -0,0 +1,228 @@
+---
+abstract: This specification defines a way for an XMPP entity to
+  announce the limits it will enforce for data received on a stream.
+approver: Council
+author:
+- "Kim Alvefur <zash@zash.se> <xmpp:zash@zash.se>"
+- "Matthew Wild <mwild1@gmail.com> <xmpp:me@matthewwild.co.uk>"
+date: 2022-10-20
+number: xxxx
+revision:
+- date: 2021-08-20
+  initials: ka
+  remark:
+  - Early draft
+  version: 0.0.1
+- date: 2021-08-25
+  initials: ka
+  remark:
+  - more work
+- date: 2021-08-28
+  initials: ka
+  remark:
+  - more words
+- date: 2022-06-01
+  initials: ka
+  remark:
+  - word tweaks
+  - element name and namespace un-TDB'd
+- date: 2022-10-20
+  initials: ka, mw
+  remark:
+  - First draft.
+  version: 0.0.1
+shortname: sla
+sig: Standards
+status: ProtoXEP
+title: "XEP-xxxx: Stream Limits Advertisement"
+type: Standards Track
+---
+
+# Introduction {#intro}
+
+This documents describes a mechanism for communicating limits, such as
+stanza size limits that is in effect on a particular stream, in order to
+allow the sending party to avoid reaching those limits.
+
+## Problem statement {#intro-problem}
+
+Where stanza size limits have been deployed, very often this leads to
+problems with large stanzas causing connection outages, most often [User
+Avatar (XEP-0084)](https://xmpp.org/extensions/xep-0084.html) [^1] and
+[XMPP Registrar Function
+(XEP-0053)](https://xmpp.org/extensions/xep-0053.html) [^2] result
+stanzas, which can be very large due to embedded images.
+
+If stanza size limit violations are met with stream errors then this may
+lead to temporary connection outage, which may a few seconds to recover
+from.
+
+# Requirements {#reqs}
+
+-   Enable discovery of the stanza size limit in use on a stream.
+
+-   Support for bi-directional streams.
+
+These requirements will enable XMPP clients and servers to adapt data
+they generate, such that it will fit within the limits required by the
+recipient, or reject overly large stanzas early, rather than following a
+trial-and-error approach.
+
+# Use Cases {#usecases}
+
+## Server advertises limits to connecting entity {#advertise}
+
+For any XMPP stream, there is an \"initiating entity\" (a client or
+server) and a \"responding entity\" that they are connecting to. The
+responding entity advertises its limits in the \<stream:features/\>
+element that it sends at the start of the stream.
+
+The limits are enclosed in a \<limits/\> element qualified by the
+\'urn:xmpp:stream-limits:0\' namespace. This element SHOULD contain the
+following child elements:
+
+\<max-bytes/\>
+:   Contains an integer representing the maximum size of any first-level
+    stream elements (including stanzas), in bytes. Guidance on
```

Comment:
While reading this, I was wondering in which direction this limit is enforced. Is it the maximum the announcing entity is willing to accept or is going to send. I think only the former makes sense, so maybe clarify by

```suggestion
    stream elements (including stanzas), in bytes the announcing entity is willing to accept. Guidance on
```